Hot Reload Development Environment and Dockerize app

ilmimris opened the pull request:
Untuk test jalanin servicenya, pakai command `make dev`.
command `make dev` akan setup config terlebih dahulu, baru build docker image lalu run docker container.

Commit:
Merge pull request #2 from iwanjunaid/main

sync

iwanjunaid committed:
Refactor rest.

iwanjunaid committed:
Merge branch 'main' into simplify-rest-endpoints

iwanjunaid committed:
Query with context.

iwanjunaid committed:
Remove unused fmt statement.

ilmimris committed:
wip: dockerize basesvc

Commit:
Merge pull request #3 from iwanjunaid/simplify-rest-endpoints

Simplify rest endpoints

ilmimris committed:
feat: wip: hot reload for dev env

ilmimris committed:
Merge branch 'hotreload' of github.com:ilmimris/basesvc into hotreload

wibiaris committed:
group routing

wibiaris committed:
initial data store

ilmimris committed:
feat: wip: hotreload docker

ilmimris committed:
feat: wip: hotreload update docker config

ilmimris committed:
clean up test change description

ilmimris committed:
build: add make option run and stop

- run for start container using docker compose
- stop for take down container

ilmimris committed:
docs: add docs

ilmimris committed:
ignore tmp dir and app binary

iwanjunaid committed:
- Remove version command
- Add swagger documentation
- Add /v2 endpoints

wibiaris committed:
initial data store

iwanjunaid committed:
Ignore bin/ dir.

iwanjunaid committed:
Change build output folder.

iwanjunaid committed:
Merge pull request #5 from iwanjunaid/simplify-rest-endpoints

Simplify rest endpoints

Commit:
refactoring, add hot reload config change and graceful shutdown.

wibiaris committed:
initial config mongodb, redis & configurable service optional source

wibiaris committed:
fix conflict

wibiaris committed:
remove router folder

ilmimris committed:
change default binding port

ilmimris committed:
change `make dev` command

when `make dev` execute it will execute `make setup` then `make run`

Commit:
Merge branch 'develop' into develop

ilmimris committed:
fix missing git

ilmimris committed:
handle when bin/swag not available

wibiaris commented on cmd/rest.go:
```diff
@@ -1,21 +1,53 @@
+			cfg = config.GetConfig()
+
+			// database configuration
+			db = datastore.NewDB(cfg)
```
taro di bootstrap aja initial data source, klo ada another command gak perlu di load lagi seperti ini

iwanjunaid replied:
ya setuju, initialisasi yg dibutuhkan command lain, harusnya diletakkan di bootstrap

wibiaris commented on cmd/rest.go:
```diff
@@ -1,21 +1,53 @@
+			db = datastore.NewDB(cfg)
+
+			// logging configuration
+			logConfig := logger.Configuration{
```
logging juga tempatnya di bootstrap aja

wibiaris commented on config/db/mysql.go:
```diff
@@ -0,0 +1,24 @@
+}
+
+type Database struct {
+	User                 string `yaml:"username"`
```
config file jadi pakai yaml ?

ilmimris replied:
Jadi pakai yaml

wibiaris commented on cmd/version.go:
```diff
@@ -0,0 +1,15 @@
+	"github.com/spf13/cobra"
+)
+
+var versionCmd = &cobra.Command{
```
di upstream sudah dihapus, coba di pull main yang terbaru

ilmimris replied:
Oke

iwanjunaid committed:
Merge pull request #6 from wibiaris/feature/rest-entry-point

Feature/rest entry point

ilmimris committed:
change requirement modules

ilmimris committed:
dockerize Prod and Dev env

ilmimris committed:
dockerize prod:

- add `prod` and `run-prod`
- remove `make setup` inside prod Dockerfile
- update docker-compose config dev and prod

ilmimris committed:
sync with iwanjuaid/basesvc main branch

ilmimris committed:
Merge branch 'main' of github.com:iwanjunaid/basesvc into main

Commit:
add logger

Commit:
removing config.yaml

ilmimris committed:
Merge remote-tracking branch 'origin/feat/logger' into develop

ilmimris committed:
adopt config style in branch feat/logger

ilmimris committed:
fix dependency error

Commit:
removing all yml files.

ilmimris committed:
Merge branch 'feat/logger' into develop

ilmimris committed:
delete all yaml

ilmimris committed:
move hot reload config into root dir

ilmimris committed:
ignore .env and basesvc.config.json

ilmimris committed:
update make setup config

ilmimris committed:
update example config.json

ilmimris committed:
change cmd docker

ilmimris committed:
fix wrong constant for mongo config

ilmimris committed:
handle case when .env and basesvc.config.json exist in local then not copy from .dist

ilmimris committed:
update docs
